examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,247 @@
+# The physical parameters are
+#
+#   * ``f``: Coriolis parameter
+#   * ``g``: Acceleration due to gravity
+#   * ``U``: Maximum jet speed
+#   * ``\Delta\eta``: Maximum free-surface deformation that is dictated by geostrophy
+#   * ``\epsilon`` : Amplitude of the perturbation
+
+f = 1
+g = 9.80665
+U = 1.0
+Δη = f * U / g
+ϵ = 1e-4
+nothing # hide
+
+# ## Building a `ShallowWaterModel`
+#
+# We use `grid`, `coriolis` and `gravitational_acceleration` to build the model.
+# Furthermore, we specify `RungeKutta3` for time-stepping and `WENO5` for advection.
+
+model = ShallowWaterModel(
+    timestepper=:RungeKutta3,
+    advection=WENO5(),
+    grid=grid,
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions have a small perturbation that is random in space and 
+# decays away from the center of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2)
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 + ϵ * exp(- (y - Ly/2)^2 ) * randn()
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2)
+ hⁱ(x, y, z) = model.grid.Lz + ηⁱ(x, y, z)
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z)
+nothing # hide
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
+   ω_pert = ComputedField( ω_field - Ω )
+
+# Progress will output the clock times and the norm of the cross channel velocity,
+# as this is used to compute the growth rates.  We obtain the `norm` function from `LinearAlgebra`.
+
+using LinearAlgebra, Printf
+
+function progress(sim)
+    compute!(v)
+    @printf("Iteration: %d, time: %s, norm v: %f\n",
+    sim.model.clock.iteration,
+    prettytime(sim.model.clock.time),
+    norm(interior(v)) )
+end
+nothing # hide
+
+# ## Running a `Simulation`
+#
+# We pick the time-step that ensures to resolve the surface gravity waves.
+# A time-step wizard can be applied to use an adaptive time step.
+
+simulation = Simulation(model, Δt = 1e-2, stop_time = 150.0, progress=progress)
+
+# ## Prepare output files
+#
+# Define a function to compute the growth rate based on the cross channel velocity
+
+function growth_rate(model)
+    compute!(v)
+    return norm(interior(v))
+end
+nothing # hide
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
```

Comment:
wouldn't `simulation.output_writers[:fields].filepath` be simpler instead of `joinpath(@__DIR__, "Bickley_jet_shallow_water.nc")`?

Reply:
This is something that @ali-ramadhan helped me with as when I first ran it it could not find my data file.  I will try it without this to see if things work.  I would prefer to not have it if possible.

Reply:
both @ali-ramadhan  and I have found that we need this in order for the docs to build.  So I will keep it for now and maybe we can sort out why in the future?